File: Mod3/Clase09/app/src/main/java/com/area51/clase09/RegistroActivity.java
package com.area51.clase09;

import androidx.appcompat.app.AppCompatActivity;

import android.location.Address;
import android.location.Geocoder;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.area51.clase09.database.MetodoSQL;

import java.io.IOException;
import java.util.List;

public class RegistroActivity extends AppCompatActivity {
    private EditText etDireccion, etLatitud, etLongitud;
    private Button btnBuscar, btnGuardar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_registro);

        etDireccion = findViewById(R.id.etDireccion);
        etLatitud = findViewById(R.id.etLatitud);
        etLongitud = findViewById(R.id.etLongitud);
        btnBuscar = findViewById(R.id.btnBuscar);
        btnGuardar = findViewById(R.id.btnGuardar);

        btnBuscar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String direccion = etDireccion.getText().toString();

                Geocoder geocoder = new Geocoder(RegistroActivity.this);
                try {
                    List<Address> direcciones =
                            geocoder.getFromLocationName(direccion, 1);
                    if (direcciones != null &&
                            direcciones.size() > 0) {
                        Address dire = direcciones.get(0);
                        etLatitud.setText(String.valueOf(dire.getLatitude()));
                        etLongitud.setText(String.valueOf(dire.getLongitude()));
                    }
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        });
        btnGuardar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String direccion = etDireccion.getText().toString();
                String latitud = etLatitud.getText().toString();
                String longitud = etLongitud.getText().toString();

                MetodoSQL metodoSQL = new MetodoSQL(RegistroActivity.this);
                boolean seGrabo = metodoSQL.agregar(direccion,
                        Double.parseDouble(latitud),
                        Double.parseDouble(longitud));
                if (seGrabo) {
                    Toast.makeText(RegistroActivity.this,
                            "Se grabo la ubicación", Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(RegistroActivity.this,
                            "No se grabo", Toast.LENGTH_SHORT).show();
                }

            }
        });
    }
}

File: Mod3/Clase09_1/app/src/main/java/com/area51/clase09_1/RegistroActivity.kt
package com.area51.clase09_1

import android.location.Geocoder
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.area51.clase09_1.database.MetodoSQL
import kotlinx.android.synthetic.main.activity_registro.*

class RegistroActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_registro)

        btnAgregar.setOnClickListener {
            val direccion = etDireccion.text.toString()
            val latitud = etLatitud.text.toString()
            val longitud = etLongitud.text.toString()

            val metodoSQL = MetodoSQL(this)
            val resultado = metodoSQL.agregar(
                direccion,
                latitud.toDouble(), longitud.toDouble()
            )
            if (resultado) {
                Toast.makeText(
                    this, "se registro", Toast.LENGTH_SHORT
                ).show()
            } else {
                Toast.makeText(
                    this, "No se registro", Toast.LENGTH_SHORT
                ).show()
            }
            //https://sqlitebrowser.org
        }
        btnBuscar.setOnClickListener {
            val direccion = etDireccion.text.toString()

            val geocoder = Geocoder(this)
            val direcciones = geocoder.getFromLocationName(direccion, 1)
            if (direcciones != null &&
                direcciones.size > 0
            ) {
                val dir = direcciones[0]

                etLatitud.setText(dir.latitude.toString())
                etLongitud.setText(dir.longitude.toString())
            }
        }
    }
}
